### PR TITLE
Add application payload formatters e2e tests

### PR DESCRIPTION
**Pull request by asmulko**
#### Summary

References #3606 

#### Changes

- Added application payload formatters e2e tests
- Added end device payload formatters e2e tests
- Added `toast` notification for end device downlink (use application payload formatter option) save success
- Renamed several tests according to testing guidelines

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by asmulko** [comment]
Work in progress, I am going to add end devices payload formatters here as well

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+  })
+
+  describe('Application', () => {
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createApplication(application, userId)
+      cy.clearLocalStorage()
+      cy.clearCookies()
```
Comment (kschiffer) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: Is this really necessary? Cypress should clear cookies and localStorage between tests anyway.

[https://docs.cypress.io/api/commands/clearlocalstorage.html#Syntax](https://docs.cypress.io/api/commands/clearlocalstorage.html#Syntax)

https://docs.cypress.io/api/commands/clearcookies.html#Syntax

See also my comment below. I think we can remove this before hook altogether.

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+    })
+
+    describe('Uplink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+      })
```
Comment (kschiffer) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: I think we should avoid these double logins. They only take time and don't provide any value. You could rather login and create the application/end device in these before hooks.

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+
+      it('successfully changes formatter type to Javascript', () => {
+        cy.visit(
+          `${Cypress.config(
+            'consoleRootPath',
+          )}/applications/${applicationId}/payload-formatters/uplink`,
+        )
+
+        cy.findByLabelText('Javascript').check()
+        cy.findByTestId('code-editor').should('exist')
```
Comment (kschiffer) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]:
```suggestion
        cy.findByTestId('code-editor').should('be.visible')
```

**Review by kschiffer — CHANGES_REQUESTED** [review]
Have some small remarks.

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    describe('Uplink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+      })
+
+      it('successfully changes formatter type to Javascript', () => {
```
Comment (bafonins) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]:
```suggestion
      it('succeeds changing formatter type to Javascript', () => {
```

Please revisit our testing guidelines https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.11/DEVELOPMENT.md#javascript-based-tests

**Review by bafonins — CHANGES_REQUESTED** [review]

File: cypress/integration/console/organizations/edit.spec.js [hunk 5]
```diff
@@ -39,7 +39,7 @@ describe('Organization general settings', () => {
     )
   })
 
-  it('successfully edit organization name and description', () => {
+  it('succeed editing organization name and description', () => {
```
Comment (bafonins) [inline cypress/integration/console/organizations/edit.spec.js]:
```suggestion
  it('succeeds editing organization name and description', () => {
```

File: cypress/integration/console/organizations/edit.spec.js [hunk 6]
```diff
@@ -52,7 +52,7 @@ describe('Organization general settings', () => {
       .should('be.visible')
   })
 
-  it('successfully delete organization', () => {
+  it('succeed deleting organization', () => {
```
Comment (bafonins) [inline cypress/integration/console/organizations/edit.spec.js]:
```suggestion
  it('succeeds deleting organization', () => {
```

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+    cy.createUser(user)
+    cy.loginConsole({ user_id: userId, password: user.password })
+    cy.createApplication(application, userId)
+    cy.createEndDevice(applicationId, endDeviceRequestBody)
+  })
+
+  describe('Application', () => {
+    describe('Uplink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
```
Comment (bafonins) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: nit
```suggestion
        cy.loginConsole({ user_id: userId, password: user.password })
```

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 8]
```diff
@@ -0,0 +1,506 @@
...
+        cy.findByTestId('toast-notification')
+          .should('be.visible')
+          .findByText('Payload formatter updated')
+          .should('be.visible')
+      })
+    })
+
+    describe('Downlink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
```
Comment (bafonins) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: nit
```suggestion
        cy.loginConsole({ user_id: userId, password: user.password })
```

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 9]
```diff
@@ -0,0 +1,506 @@
...
+          .findByText('Payload formatter updated')
+          .should('be.visible')
+      })
+    })
+  })
+
+  describe('End Devices', () => {
+    describe('Uplink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
```
Comment (bafonins) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: nit
```suggestion
        cy.loginConsole({ user_id: userId, password: user.password })
```

File: cypress/integration/console/shared/payload-formatters/edit.spec.js [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+        cy.findByTestId('toast-notification')
+          .should('be.visible')
+          .findByText('Payload formatter updated')
+          .should('be.visible')
+      })
+    })
+
+    describe('Downlink', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
```
Comment (bafonins) [inline cypress/integration/console/shared/payload-formatters/edit.spec.js]: nit
```suggestion
        cy.loginConsole({ user_id: userId, password: user.password })
```

File: pkg/webui/console/containers/device-payload-formatters/downlink.js [hunk 11]
```diff
@@ -100,6 +100,16 @@ class DevicePayloadFormatters extends React.PureComponent {
     })
   }
 
+  @bind
+  async onSubmitSuccess() {
```
Comment (bafonins) [inline pkg/webui/console/containers/device-payload-formatters/downlink.js:98]: Please remove `L96-L100` then. Currently, two toasts are shown on successful payload formatter update.

Comment (bafonins) [reply pkg/webui/console/containers/device-payload-formatters/downlink.js:98]: You can use the uplink form as a reference

**Review by bafonins — CHANGES_REQUESTED** [review]
Please see my comments

**Review by bafonins — APPROVED** [review]

**Review by kschiffer — APPROVED** [review]

**Pull request merged**